flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java

```diff
@@ -9,28 +9,93 @@
 import com.android.tools.idea.npw.module.ModuleGalleryEntry;
 import com.android.tools.idea.npw.module.NewModuleModel;
 import com.android.tools.idea.wizard.model.SkippableWizardStep;
-import com.intellij.openapi.diagnostic.Logger;
 import icons.FlutterIcons;
 import io.flutter.FlutterBundle;
+import io.flutter.project.FlutterProjectModel;
+import io.flutter.project.FlutterProjectStep;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import javax.swing.*;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 public class FlutterDescriptionProvider implements ModuleDescriptionProvider {
-  private static final Logger LOG = Logger.getInstance(FlutterDescriptionProvider.class.getName());
 
-  @Override
-  public Collection<? extends ModuleGalleryEntry> getDescriptions() {
+  public static List<ModuleGalleryEntry> getGalleryList(boolean includeApp) {
     ArrayList<ModuleGalleryEntry> res = new ArrayList<>();
+    if (includeApp) {
+      res.add(new FlutterApplicationGalleryEntry());
+    }
     res.add(new FlutterPluginGalleryEntry());
     res.add(new FlutterPackageGalleryEntry());
     return res;
   }
 
-  private static class FlutterPackageGalleryEntry implements ModuleGalleryEntry {
+  @Override
+  public Collection<? extends ModuleGalleryEntry> getDescriptions() {
+    return getGalleryList(false);
+  }
+
+  /**
+   * Each type of Flutter project is represented by a class that implements this interface.
+   */
+  public interface FlutterGalleryEntry extends ModuleGalleryEntry {
+
+    // TODO: Unify module and project creation so we don't need this method.
```

Comment:
I'd use the `// TODO(stevemessick): ...` format, so people later know who has the most context for this todo.